portal-api/src/v1/controllers/utilisation-report-service/utilisation-report-upload.controller.js

```diff
@@ -1,161 +1,80 @@
     const { file } = req;
 
-    const { formattedReportPeriod, reportData, reportPeriod, user } = req.body;
-    const parsedReportData = JSON.parse(reportData);
-    const parsedUser = JSON.parse(user);
-    const parsedReportPeriod = JSON.parse(reportPeriod);
-
     if (!file) {
       return res.status(400).send();
     }
 
+    const { formattedReportPeriod, reportData, reportPeriod, user } = req.body;
+    const parsedReportData = JSON.parse(reportData);
+    const parsedUser = JSON.parse(user);
+    const parsedReportPeriod = JSON.parse(reportPeriod);
     const bankId = parsedUser?.bank?.id;
 
-    const existingReports = await api.getUtilisationReports(bankId, {
-      reportPeriod: parsedReportPeriod,
-    });
-
-    if (existingReports.length !== 1) {
-      return res.status(500).send(`Expected 1 report but found ${existingReports.length} with bank ID ${bankId} and report period '${parsedReportPeriod}'`);
-    }
-
-    const existingReport = existingReports[0];
-
-    if (existingReport.status !== UTILISATION_REPORT_RECONCILIATION_STATUS.REPORT_NOT_RECEIVED) {
-      return res
-        .status(500)
-        .send(
-          `Expected report to be in '${UTILISATION_REPORT_RECONCILIATION_STATUS.REPORT_NOT_RECEIVED}' state (was actually in '${existingReport.status}' state)`,
-        );
+    const report = await getReportForPeriod(bankId, parsedReportPeriod);
+    validateReportIsInReportNotReceivedState(report);
+    const azureFileInfo = await saveUtilisationReportFileToAzure(file, bankId);
+    const { dateUploaded } = await api.saveUtilisationReport(report.reportId, parsedReportData, parsedUser, azureFileInfo);
+
+    try {
+      await sendUtilisationReportUploadNotificationEmailToUkefGefReportingTeam(
+        parsedUser?.bank?.name,
+        formattedReportPeriod,
+      );
+    } catch (error) {
+      console.error('Failed to send report upload notification to ukef gef reporting team %o', error);
     }
-
-    const fileInfo = await saveFileToAzure(file, bankId);
-
-    const azureFileInfo = {
-      ...fileInfo,
-      mimetype: file.mimetype,
-    };
-
-    const saveDataResponse = await api.saveUtilisationReport(existingReport.id, parsedReportData, parsedUser, azureFileInfo);
-
-    await sendEmailToPdcInputtersEmail(parsedUser?.bank?.name, formattedReportPeriod);
-    const { paymentOfficerEmails } = await sendEmailToBankPaymentOfficerTeam(
+    const { paymentOfficerEmails } = await sendUtilisationReportUploadConfirmationEmailToBankPaymentOfficerTeam(
       formattedReportPeriod,
       parsedUser?.bank?.id,
-      new Date(saveDataResponse.dateUploaded),
-      parsedUser,
+      new Date(dateUploaded),
+      parsedUser.firstname,
+      parsedUser.surname,
     );
     return res.status(201).send({ paymentOfficerEmails });
   } catch (error) {
-    console.error('Failed to save utilisation report %o', error);
+    if (error instanceof InvalidReportStatusError) {
+      return res.status(500).send(error.message);
```

Comment:
Could we update `InvalidReportStatusError` to have its own error code rather than hard-coding the `500` here?